rskj-core/src/test/java/co/rsk/peg/bitcoin/BitcoinUtilsTest.java

```diff
@@ -198,46 +247,121 @@ void getFirstInputSigHash_forP2shP2wshTx_withNoRedeemScript_shouldReturnEmpty()
-        Script redeemScript = new Script(redeemScriptHex);
+        int outputIndex = 0;
+        int nHash = 0;
+        Federation segwitCompatibleFederation = P2shP2wshErpFederationBuilder.builder().build();
+        Script emptyScript = new Script(new byte[]{});
+        Script redeemScript = segwitCompatibleFederation.getRedeemScript();
+        btcTx.addInput(BitcoinTestUtils.createHash(nHash), outputIndex, emptyScript);
+
+        TransactionWitness witness = createBaseWitnessThatSpendsFromRedeemScript(redeemScript);
+        btcTx.setWitness(FIRST_INPUT_INDEX, witness);
+
+        Coin minimumPegoutTxValue = bridgeMainnetConstants.getMinimumPegoutTxValue();
+        btcTx.addOutput(minimumPegoutTxValue, destinationAddress);
+        btcTx.addOutput(minimumPegoutTxValue, segwitCompatibleFederation.getAddress());
 
         // Act
-        Optional<Sha256Hash> sigHash = getFirstInputSigHash(btcTx);
+        Optional<Script> redeemScriptFromInput = BitcoinUtils.extractRedeemScriptFromInput(btcTx, FIRST_INPUT_INDEX);
 
         // Assert
-        assertTrue(sigHash.isPresent());
-        Sha256Hash expectedSigHash = btcTx.hashForSignature(
-            0,
-            redeemScript,
-            BtcTransaction.SigHash.ALL,
-            false
-        );
-        assertEquals(expectedSigHash, sigHash.get());
+        assertTrue(redeemScriptFromInput.isPresent());
+        assertEquals(redeemScript, redeemScriptFromInput.get());
+    }
+
+
+    @Test
+    void extractRedeemScriptFromInput_withAMigrationTxWithMultipleP2shP2wshInputs_shouldExtractThemProperly() {
+        // Arrange
+        BtcTransaction migrationTx = new BtcTransaction(btcMainnetParams);
+
+        int outputIndex = 0;
+        Federation retiringFederation = P2shP2wshErpFederationBuilder.builder().build();
+        Script retiringFedRedeemScript = retiringFederation.getRedeemScript();
+        Script emptyScript = new Script(new byte[]{});
+
+        Federation activeFederation = P2shP2wshErpFederationBuilder.builder().build();
+        Address activeFederationAddress = activeFederation.getAddress();
+
+        int numberOfInputAndOutputs = 3;
```

Comment:
```suggestion
        int numberOfInputs = 3;
```
just one output right? :)

Reply:
done!